validation/citrus-validation-xml/src/main/java/org/citrusframework/validation/xml/XmlMarshallingValidationProcessor.java

```diff
@@ -151,7 +151,7 @@ public XmlMarshallingValidationProcessor<T> build() {
                         "please add proper validation logic");
             }
 
-            return new XmlMarshallingValidationProcessor<>() {
+            return new XmlMarshallingValidationProcessor<>(unmarshaller) {
```

Comment:
could you please describe the bug and the solution to me? I don't get it just from looking at this, sorry 😢

Reply:
I think I understand the issue. The issue is that the builder does not set the unmarshaller on the validation processor when building the instance. As a result the unmarshaller is always resolved at runtime via reference resolver lookup and setting a custom unmarshaller on the builder is not possible. Good fix! Thanks a lot